### ISSUE TITLE
Contrastive loss of generator and msp module do not converge

### ISSUE THREAD

**Issue by yz-mo** [zyxElsa/CAST_pytorch#10]
Hi~ I'm trying to do something like CAST. But in my network, the contrastive loss of generator and msp module do not converge, and the value of these two losses are very similar, have you ever encountered a similar situation? The loss curves is as follows, and they are still the same when training is done (200epoch).
![屏幕截图 2022-08-17 103731](https://user-images.githubusercontent.com/108313059/185022363-a2884d41-005f-46af-a878-ad505bdfcb68.png)

**Comment by zyxElsa** [comment]
Hi~I did not encounter this situation. 
Here are some of my guesses, hope they are useful to you:
1. Contrastive-related parameters such as temperature (0.07 in CAST), or queue length, i.e. the number of negative samples (4096 in CAST) can be adjusted.
2. The weight of the contrastive loss when training G can be adjusted (0.2 in CAST).
3. The learning rate and descent strategy can be changed.
4. Double-check the related  codes.

**Comment by yz-mo** [comment]
Thanks very much! I adjusted the setting of the query, key, and negative samples. Though the losses still do not converge, the testing results are much better than before.
